data_structures/src/data_request.rs

```diff
@@ -343,38 +343,39 @@ pub fn create_tally<RT>(
     dr_output: &DataRequestOutput,
     pkh: PublicKeyHash,
     report: &RadonReport<RT>,
-    reveals: Vec<RevealTransaction>,
-    n_commits: usize,
+    reveal_pkhs: Vec<PublicKeyHash>,
+    commit_pkhs: Vec<PublicKeyHash>,
 ) -> Result<TallyTransaction, failure::Error>
 where
     RT: TypeLike,
 {
     if let Stage::Tally(tally_metadata) = &report.metadata {
         let reveal_reward = dr_output.witness_reward;
+        let n_commits = commit_pkhs.len();
+        let mut slashed_witnesses: HashSet<PublicKeyHash> = commit_pkhs.iter().cloned().collect();
 
         let liars = &tally_metadata.liars;
 
-        if reveals.len() != liars.len() {
+        if reveal_pkhs.len() != liars.len() {
             return Err(TransactionError::MismatchingLiarsNumber {
-                reveals_n: reveals.len(),
+                reveals_n: reveal_pkhs.len(),
                 inputs_n: liars.len(),
             }
             .into());
         }
 
-        let mut rewarded_witnesses = vec![];
-        let mut outputs: Vec<ValueTransferOutput> = reveals
+        let mut outputs: Vec<ValueTransferOutput> = reveal_pkhs
             .iter()
             .zip(liars.iter())
-            .filter_map(|(reveal, &liar)| {
+            .filter_map(|(&reveal_pkh, &liar)| {
```

Comment:
In accordance to previous suggestions:
```suggestion
            .filter_map(|(&revealer, &liar)| {
```

You will see a line down below become super expressive:
```rust
slashed_witnesses.remove(&revealer);
```